pkg/features/feature.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import "k8s.io/apimachinery/pkg/util/sets"
+
+// -----------------------------------------------------------------------------
+// Features - Types
+// -----------------------------------------------------------------------------
+
+// FeatureName allows opting in to additional conformance tests at an
+// individual feature granularity.
+type FeatureName string
+
+type FeatureStatus string
+
+// TODO: comment
+const (
+	FeatureStatusTrial  = "TRIAL"
```

Comment:
The more I think about this, the more I think that we actually need two separate levels here:

1. The stability of the feature (the max release channel this has made it to)
2. The stability of the test (tests that are brand new should default to a lower level of stability than tests that have been around for a release cycle or two with multiple passing implementations)

I think it's useful to solve both of these, and this PR seems to be on a path to solve the first one. With that said, if we're agreeing that this PR is primarily targeting the first item, then I'd use the existing "Experimental" and "Standard" terminology for each feature.

Solving 2 may actually be easier. I think it just becomes a marker on tests, and implementations have the option to skip these tests when generating a report without any kind of penalty in their conformance report (or something along those lines).

Reply:
Yep, I agree with the distinction and the fact that this PR is suitable to address the first point. As you pointed out, the 2nd point is easier, as it can be done via a specific field on the test struct.

I'll change the feature status names accordingly.

Reply:
I've done the change and marked the PR as ready for review, PTAL :)